src/playlists.ts

```diff
@@ -0,0 +1,37 @@
+import * as vscode from 'vscode';
+
+
+
+export class PlaylistsProvider implements vscode.TreeDataProvider<Playlist> {
+  constructor(private spotify: any) {}
+
+
+  getChildren(element?: Playlist): Thenable<Playlist[]> {
+    if (element) {
+      return Promise.resolve([]);
+    }
+
+    const user = this.spotify.getMe();
+
+    return Promise.resolve(this.spotify.getUserPlaylists(user.id)
+    .then((data) => {
+      return Promise.resolve(data.body.items.map((playlist: any) => new Playlist(playlist.name)));
+    },function(err) {
+      console.log('Something went wrong!', err);
+    })
+  );
+  }
+
+  getTreeItem(element: Playlist): vscode.TreeItem {
+    return element;
+  }
+
+}
+
+export class Playlist extends vscode.TreeItem {
+  constructor (public readonly name: string) {
+    super(name);
+    this.description = name;
+    this.tooltip = new vscode.MarkdownString(`name`);
+  }
+}
```

Comment:
Alright, let me explain what was going on in my original code (copied below)
```typescript
this.tooltip = new vscode.MarkdownString(`**${this.name}**  \n${this.artists.map(x => x.name).join(', ')}`);
```

The backticks mean that the string I'm shoving in is a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals), which lets me embed JavaScript expressions/variables inside the string without having to do any of that <kbd>+</kbd> garbage. The stuff inside the `${}` are executed as JavaScript.

I'm also using `vscode.MarkdownString` because I wanted to add a little formatting (bolding the title):
![image](https://user-images.githubusercontent.com/11380394/108014646-f4c0f100-6fd3-11eb-8656-46df1564e4cd.png)

You're not doing any special text formatting here and you're not building out a string, so you don't need either `MarkdownString` or template literals and could just do
```ts
this.tooltip = name;
```

However, if you don't specify `TreeItem.tooltip` it defaults to the value of `TreeItem.description`. Since you're setting both to the same value, you don't really need to set `tooltip` in the first place. Also this isn't mentioned explicitly in the **absolutely stunning** VS Code API reference so don't feel bad about not knowing this.

```suggestion
  }
}

```